DESIGN.md

```diff
@@ -0,0 +1,47 @@
+# HTTP Provider Design
+
+The HTTP Provider offers functionality for interacting with generic HTTP servers as part of terraform configuration.
+Specifically, the provider issues an HTTP GET request to the defined URL on every Terraform run.
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide managed data source to issue HTTP GET request. The underlying default
+[transport](https://pkg.go.dev/net/http#Transport) uses [HTTP/1.1](https://datatracker.ietf.org/doc/html/rfc2616) for
+HTTP URLs and either [HTTP/1.1](https://datatracker.ietf.org/doc/html/rfc2616) or 
+[HTTP/2.0](https://datatracker.ietf.org/doc/html/rfc7540) for HTTPS URLs depending on whether the server supports
```

Comment:
Nit: Maybe just me but the usage of "URLs" tripped me up slightly here, as I thought it was somehow relating to the actual URL (scheme, host, port, etc.) Maybe clearer to say some thing like HTTP (plaintext) requests and HTTPS (secure) requests?

Reply:
Fair point. Have amended.